File: 2. python/1. basics/lecture 2/p1.py
class Solution:
    def __init__(self, n): #constructor, self = this ptr
        for i in range(n):
            for j in range(n):
                print("* ", end="") # 'end' to control print in next lineby default
            print()

''' pattern output:

* * * *
* * * *
* * * *
* * * *

'''

if __name__ == "__main__":
    size = int(input("enter size: "))

    print("1. output:")
    Solution(size)

File: 3. JS/1. basics/lecture 2/p1.ts
import * as readline from 'readline';

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

class Solution {
  constructor(n: number) {
    for (let i = 1; i <= n; i++) {
      let row: string = "";
      for (let j = 1; j <= n; j++) {
        row += "* ";
      }
      console.log(row);
    }
  }
}

/* pattern output:

* * * *
* * * *
* * * *
* * * *

*/

console.log("Enter size of pattern:");

rl.on('line', (input: string) => {
  const size: number = parseInt(input);
  
  console.log("directly constructor based output:");
  new Solution(size);
  
  rl.close();
});